src/wallet/wallet.cpp

```diff
@@ -1330,7 +1330,36 @@ void CWallet::SyncVoidedCert(const uint256& certHash, bool bwtAreStripped)
     CWalletDB walletdb(strWalletFile, "r+", false);
     if (!itCert->second->WriteToDisk(&walletdb))
         LogPrintf("%s():%d - ERROR in writing to db\n", __func__, __LINE__);
+}
+
+void CWallet::SyncSidechain(const uint256& scId, const CMinimalSidechain& walletSidechainData)
+{
+    CWalletDB walletdb(strWalletFile, "r+", false);
 
+    if (!walletSidechainData.IsNull())
+    {
+        mapSidechains[scId] = walletSidechainData;
+        walletdb.WriteSidechain(scId, walletSidechainData);
+    } else {
+        mapSidechains.erase(scId);
+        walletdb.EraseSidechain(scId);
+    }
+
+    return;
+}
+
+bool CWallet::ReadSidechain(const uint256& scId, CMinimalSidechain& sidechain)
+{
+    bool res = false;
+    if (mapSidechains.count(scId) != 0) {
+        sidechain = mapSidechains.at(scId);
+        res = true;
+    } else {
+        CWalletDB walletdb(strWalletFile, "r+", false);
+        res = walletdb.ReadSidechain(scId, sidechain);
```

Comment:
if res is false can we rely on retrieved value?

Reply:
No, we cannot. Fixed